osbuild/mounts.py

```diff
@@ -125,9 +125,20 @@ def __init__(self, args):
 
     # pylint: disable=no-self-use
     @abc.abstractmethod
-    def translate_options(self, options: Dict) -> List:
-        if options.get("readonly", False):
-            return ["-o", "ro"]
+    def translate_options(self, options: Dict) -> List[str]:
+        opts: List[str] = []
+        for key, value in options.items():
+            if value is True:
+                # For backward compatibility, we must keep the readonly keyword as the parameter was already
+                # existing before this fix and thus might be used oustide our control.
+                if key in "readonly":
+                    opts += ["ro"]
```

Comment:
Not sure I follow the logic here.  Are we special casing `readonly` because the option is different from the key in the schema?
Maybe we shouldn't be looping through all anyway.  The convention in osbuild stages is to just read each option one at a time and treat them separately, with `get()` and defaults, like `readonly` was handled already.

See for example, the users stage:
https://github.com/osbuild/osbuild/blob/c0fb5cf90c073d25a5de852690081dd1b71b94cb/stages/org.osbuild.users#L157-L163

Reply:
Yep, the loop itself has less interest now that we aren't throwing a RuntimeException. Good for me.